keras_nlp/applications/bert.py

```diff
@@ -198,3 +199,46 @@ def get_config(self):
             }
         )
         return config
+
+
+class BertClassifier(keras.Model):
+    """Classifier model with BertEncoder."""
+
+    # TODO(jbischof): figure out initialization default
+    def __init__(self, encoder, num_classes, initializer, **kwargs):
+        super().__init__(**kwargs)
+        self.encoder = encoder
+        self.num_classes = num_classes
+        self._logit_layer = keras.layers.Dense(
+            num_classes,
+            kernel_initializer=initializer,
+            name="logits",
+        )
+
+    def call(self, inputs):
+        # Ignore the sequence output, use the pooled output.
+        pooled_output = self.encoder(inputs)["pooled_output"]
+        return self._logit_layer(pooled_output)
+
+
+def BertBase(weights=None):
+    """Factory for BertEncoder using "Base" architecture."""
+
+    model = Bert(
+        vocab_size=30522,
+        num_layers=12,
+        hidden_size=768,
+        num_attention_heads=12,
+        inner_size=3072,
+        inner_activation="gelu",
```

Comment:
do we need to supply these? if they are already default?

Reply:
I am not sure but I like the idea of the fully specified models being fully specified so that we can freely set defaults of models and layers (re offline discussion of MLM head initializer being `relu` vs `gelu`).

Reply:
@mattdangerw how do we want the default for `num_segments` for BertBase? We clearly need to expose this as a param at least to support all downstream applications? Could be a followup PR as well

Reply:
Could this be inferred from `segment_id_input`? Would be a lot less brittle if so.

Reply:
Discussed offline. Cannot be inferred from data; we'll just fix at two for now since this should still work in the n=1 case